boa/src/builtins/string/mod.rs

```diff
@@ -1165,6 +1166,81 @@ impl String {
         }
     }
 
+    /// String.prototype.split()
+    ///
+    /// The `split()` method divides a String into an ordered list of substrings, puts these substrings into an array, and returns the array.
+    ///
+    /// The division is done by searching for a pattern; where the pattern is provided as the first parameter in the method's call.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-string.prototype.split
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split
+    pub(crate) fn split(this: &Value, args: &[Value], context: &mut Context) -> Result<Value> {
+        let string = this.to_string(context)?;
+
```

Comment:
We're skipping step 1 of the spec, looking at other methods in this file it seems like we do it in many of them, we can leave the others for later, but I'd like to have this one be right.
[`RequireObjectCoercible`](https://tc39.es/ecma262/#sec-requireobjectcoercible) requires us to to throw a `TypeError` if `this` is `undefined` or `null`.

I went to check which exception is thrown by Chrome: `String.prototype.split called on null or undefined`
But `RequireObjectCoercible` is also called in some RunTimeSemantics algorithms, and in that case the messages won't have a function to reffer to. Also as an example that doesn't mention the function name, `Object.fromEntries(null)` results in `TypeError: undefined is not iterable`.

For now just writing a function that will throw a `TypeError` mentioning `null` and `undefined` seems like enough to me.

Reply:
Should we add that function to `context`, like others?

Reply:
Yeah, making it a method of `Context` might be best.

Reply:
So do we have any existing function, not limited to `context` methods, to do this? If not, is it just adding an if block and return TypeError if `this.is_null_or_undefined()` is true?

Reply:
Since this is a pattern that will repeat in a lot of functions I'd ask you to implement the method in `Context` (that can be just an `if` for now) and in `split` we would have:
```rust
let this = context.require_object_coercible(this)?
```
And the implementation of `require_object_coercible` can probably just be:
```rust
fn require_object_coercible(&self, value: &Value) -> Result<Value> {
    if value.is_null_or_undefined() {
        self.throw_type_error("value was null or undefined")
    } else {
        value
    }
}
```

Reply:
there already is a correct implementation of `RequireObjectCoercible` (`Value::require_object_coercible(context)`)